plugins/modules/nd_setup.py

```diff
@@ -464,10 +512,34 @@ def main():
             ],
         }
 
+        path = "/bootstrap/cluster"
```

Comment:
I was also thinking whether the version check is required here? In other words, will the installation process be affected in anyway if the payload consisting of ["clusterConfig"]["deploymentMode"] is pushed to an older version of ND? If it's not affected we can avoid making an api call to check the version and probably always include ["clusterConfig"]["deploymentMode"] within the payload.

Reply:
the check version might required for the path not for deployment or the new feature introduced with 3.1.1. Seems like that they change the URL path to /v2/boostrap/cluster instead of /boostrap/cluster. I am gonna try with the older path to see if it's required. I agree with you, I would love to avoid this extra API call.